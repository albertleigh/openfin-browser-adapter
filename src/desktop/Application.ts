import {BaseApiClass} from './Base';

import windowRegistry from '../helper/windowRegistry';

export class Application extends BaseApiClass{


    static staticMethods:string[]=[
        'wrap',
    ];

    static instanceMethods :string[]=[
        'addEventListener',
        'getGroups',
        'getManifest',
        'getWindow',
        'isRunning',
        'removeEventListener',
        'removeTrayIcon',
        'restart',
        'run',
        'setTrayIcon',
        'terminate',
        'wait',
    ];

    //http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#.getCurrent
    static getCurrent(){
        return new Application();
    }


    //http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#close
    close(force:boolean, callback:Function, errorCallback:Function){
        try{
            window.close();
            if(callback){
                callback();
            }
        }catch (e) {
            if (e){
                errorCallback(e);
            }
        }
    }

    //http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#getChildWindows
    getChildWindow(callback:Function, errorCallback:Function){
        if (callback){
            setTimeout(()=>{
                try{
                    callback(Object.keys(windowRegistry).map(key=>(windowRegistry[key])));
                }catch (e) {
                    if (e){
                        errorCallback(e);
                    }
                }
            },0);
        }
    }

}