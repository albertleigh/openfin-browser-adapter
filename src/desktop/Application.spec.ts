import { Application } from './Application';

describe('Desktop::Application',()=>{

    let clazz:typeof Application;

    beforeEach(()=>{
        clazz = Application;
    });

    it('all unsupported static methods defined',()=>{
        expect(clazz.staticMethods).toMatchSnapshot();
    })

    it('all unsupported instance methods defined',()=>{
        expect(clazz.instanceMethods).toMatchSnapshot();
    })

});