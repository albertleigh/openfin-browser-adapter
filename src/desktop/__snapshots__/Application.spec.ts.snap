// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop::Application all unsupported instance methods defined 1`] = `
Array [
  "getWindow",
  "addEventListener",
  "createFromManifest",
  "getGroups",
  "getInfo",
  "getManifest",
  "getParentUuid",
  "getShortcuts",
  "getTrayIconInfo",
  "getZoomLevel",
  "isRunning",
  "registerUser",
  "removeEventListener",
  "removeTrayIcon",
  "restart",
  "run",
  "scheduleRestart",
  "setShortcuts",
  "setTrayIcon",
  "setZoomLevel",
  "terminate",
]
`;

exports[`Desktop::Application all unsupported static methods defined 1`] = `
Array [
  "getCurrent",
  "wrap",
]
`;
